src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -66,8 +101,7 @@ CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
         }
     }
 
-    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
-    ReturnErrorOnFailure(reader.ExitContainer(outer));
+    chip::app::Clusters::detail::ExitStructAfterDecode(reader, outer);
```

Comment:
this loses a return if err was not CHIP_NO_ERROR.

I.e. if `reader.Next()` was returning something that is not CHIP_NO_ERROR nor CHIP_END_OF_TLV, previous code was returning that error, however new code does not.

Probably ok because the `EnterStruct` should have already done a Next pass, however it feels a bit counter-intuitive. I would make `ExitStructAfterDecode` take in and return a CHIP_ERROR.

Reply:
But in practice that will not change behavior and just be extra code....

Reply:
I was wondering how much extra code since we need CHIP_ERROR around anyway. 

I think this can be done as followups (or no at all if it turns out to be detrimental) since we seem to balance readability/maintainability vs actual space usage.